docs/decisions/0002-lerna-monorepo.rst

```diff
@@ -0,0 +1,108 @@
+
+Before further expanding the scope of what ``frontend-enterprise`` entails (i.e., adding UI components related to enterprise logistration), we should invest in splitting the domains/components of ``frontend-enterprise`` out into separate NPM packages to have better separation of concerns for consuming libraries and micro-frontend applications, reducing confusion and errors around dependencies and peer dependencies.
+
+********
+Decision
+********
+
+By separating the components of ``frontend-enterprise`` into different NPM packages, we are introducing some complexity in terms of the set up of this repository. Rather than split each domain into its own Github repository, we are taking the approach of keeping each domain in the same repository as a monorepo managed by `Lerna <https://lerna.js.org/>`_.
+
+The 3 initial packages contained in the refactored Lerna monorepo are as follows:
+
+1. ``@edx/frontend-enterprise-catalog-search``
+2. ``@edx/frontend-enterprise-logistration``
+3. ``@edx/frontend-enterprise-utils``
+
+By splitting ``frontend-enterprise`` into separate NPM packages, only the dependencies relevant to each individual package are included in the package.json file. This solves the issue of requiring peer dependencies for consuming micro-frontends to install ``frontend-enterprise`` when those peer dependencies (e.g., react-instantsearch-dom) are not relevant. While the consuming libraries and micro-frontends were already importing only the necessary code thanks to tree shaking, we can now also create a proper separation of concerns for dependencies in package.json.
+
+================================
+Packages
+================================
+
+--------------------------------
+``@edx/frontend-enterprise-catalog-search``
+--------------------------------
+
+This package exports UI components, React hooks, and actions functions related to Enterprise catalog search using `Algolia <algolia.com>`_ as a hosted search backend. This package will be used wherever it's
+necessary to allow users to search the catalog(s) available to them through their Enterprise Customer.
+
+--------------------------------
+``@edx/frontend-enterprise-logistration``
+--------------------------------
+
+This package exports a UI component related to logistration for Enterprise users. This package will be helpful for Enterprise micro-frontends to share logic related to redirecting enterprise users to the logistration page branded with their Enterprise Customer's logo and additional messaging.
+
+--------------------------------
+``@edx/frontend-enterprise-utils``
+--------------------------------
+
+This package exports useful React hooks, utility functions, and constants for use in other libraries and micro-frontends, including checking whether an authenticated user is an enterprise user based on their JWT role(s), or fetching a user's available Enterprise learner portal links.
+
+================================
```

Comment:
post this merged, I would love for this content to be part of the main README as well, it's pretty important stuff to avoid dependency confusions

Reply:
Agreed! Once packages get published, I'll be doing a quick follow-up PR to update README files to include links to things that don't exist yet until they are published.

While I'm doing that, I'll be sure to bring some of this content into the main README.